### ISSUE TITLE
TC 6.0 style transfer buggy and broken on Ubuntu cuda

### ISSUE THREAD

**Issue by romanoss** [apple/turicreate#2874]
Hi,
I tested 6.0 on my ubuntu machine with an internal RTX 2080Ti GPU and style transfer seems broken and useless.

Issues:
1) after installing in a new conda env, a tf version was installed, which disables GPU usage. Additionally installing tensorflow-gpu solved this

2) GPU memory is not freed after training

3) using batch_size param in training crashes the python kernel

4) advanced params dont work:
a) non def input_shape leads to tf error
b) print_loss_breakdown does not work
c) probably the other options dont work either

5) styling output produces similar results for different styles

6) stylized output looks bad and unusable and is absolutely not comparable to old version

7) wavy artefacts are worse than in last mx version

regards

**Comment by rainhut** [comment]
Same issues here

**Comment by romanoss** [comment]
Did you try it on a mac?

**Comment by rainhut** [comment]
Yes, the new MacBook Pro 16 Inch with an AMD Radeon Pro 5500M.

Rob

**Comment by abhishekpratapa** [comment]
@rainhut @romanoss We recently switched to another backend for style transfer and some of these got introduced. Currently investigating the bugs. Thanks for a detailed GitHub issue! Will help us track the issue down faster.

**Comment by longleb777** [comment]
you can do nothing cause the server is a beautiful hacker her name is ruby

**Comment by rainhut** [comment]
Any progress?

**Comment by abhishekpratapa** [comment]
I’m actively working on it. I’ve reproduced the problem but haven’t identified the issue yet.

**Comment by rainhut** [comment]
Thanks for the update

**Comment by abhishekpratapa** [comment]
@rainhut @romanoss I've Identified the similar results issue to be in the conditional instance normalization layer. The gradient wasn't properly being calculated through the backwards pass even though the forward pass was correct. I've fixed it in this PR: #2986

The waviness was due to loading pre-trained weights (not present in the previous model) rather than using the proper initializer. I've fixed that in this PR: #2988

More updates to come!

**Comment by romanoss** [comment]
thx for the info

**Comment by abhishekpratapa** [comment]
@rainhut @romanoss Another issue that was identified was the updatable parameters, only the transformer weights are supposed to be updated, consequently it turns out that all of the parameters of the VGG network were set to be trainable as well causing additional issues. I've fixed that in this PR: #2990

**Comment by abhishekpratapa** [comment]
@rainhut @romanoss I wasn't able to plumb through all of the advanced parameters in style transfer this release but the major blockers for training on TF have been resolved. I'll open a new issue with the remaining work for this.

**Comment by romanoss** [comment]
@abhishekpratapa thx for the info.

**Comment by rainhut** [comment]
Thanks, Abhishek. I guess we wait for the next update of TC, is that
correct?

Rob
